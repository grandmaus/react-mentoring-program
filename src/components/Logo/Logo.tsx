import React, { FC, memo } from 'react';
import { LogoLink } from './styles/LogoLink';

type Props = {
  url: string;
};

const Logo: FC<Props> = ({ url }) => (
  <LogoLink href={url}>
    <span>netflix</span>roulette
  </LogoLink>
);

export default memo(Logo);
