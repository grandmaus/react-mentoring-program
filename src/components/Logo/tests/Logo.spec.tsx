import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Logo from '../Logo';

describe('Logo', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Logo url="https://www.page.com" />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
