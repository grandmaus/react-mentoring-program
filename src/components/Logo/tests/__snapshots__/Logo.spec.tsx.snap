// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render correctly 1`] = `
<a
  className="sc-AxjAm jmUjbM"
  href="/https://www.page.com"
  onClick={[Function]}
>
  <span>
    netflix
  </span>
  roulette
</a>
`;
